### PR TITLE
Eregcsc 2290 edit button uploaded items

### PR DESCRIPTION
**Pull request by peggles2**
Resolves #EREGCSC-2290. 

**Description-**
Uploaded files related to a subject should dislay an edit button if the user has appropriate jobcodes (Eregs-admin or Eregs-manager)

**This pull request changes...**

**Backend:**
- [solution/backend/content_search/serializers.py](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-a0ea64035a8e7558f651c5e835821f2d0f4d75717668d58f9d2752f5fdca611e)
  Add an id to contentsearch serializer so that we can easily access the correct content on the admin panel.
- [solution/backend/content_search/urls.py](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-513aef3b3ceda4c39f36ea70e6a2c9f94655d3711f799f9c59ad5f4ecee907dc)
( route for the edit content search )
- [solution/backend/content_search/views.py](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-5be025c8176fc4a7cdc805e83fe4c7c3c78b92708a63bc92a38e883a00a44144)
( create edit content view so that it determines which type of content you are looking at to route you to the correct part of the admin panel) 
- [solution/backend/regulations/admin.py](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-01081b86fa9438ff17e5c4b9f22318dbaaf866df29c2db5263e58322c739c323)
Admin panel is updated so that when a user is created based on what jobcode they have they are assigned to designated user groups. These user groups will determine what access the user has. 
- [solution/backend/regulations/views/policy_repository.py](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-18af35679ffb0965fc76d4e41f485f737708c34d3a5ea1f7d233e9b5557ee9d4)
determines if the user has editable job codes or not.

**Frontend:**
- [solution/backend/regulations/templates/regulations/policy_repository.html](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-92aebe196d8a3797b0e0c929a486f9889920f68f76ff8c2fb690d57fd63c7ec6)
update the template to pass `has_editable_job_code`property to vue.js 
- [solution/ui/regulations/css/scss/partials/_policy_repository.scss](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-f25238889acc68265d186685b21736501b0433bd2574eaa1a2d8db357121f1ce)
- [solution/ui/regulations/eregs-component-lib/src/components/shared-components/ResultsItem.vue](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-8d4810ea267a137a5e8725a0d8379d192f0ef80c6e3ff51306b65d368bb2a56f)
- [solution/ui/regulations/eregs-vite/src/App.vue](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-2b9581514e99adaea419fe0430b5e1bd1fc5f870eb0eb5e42b037c396d601ec2)
- [solution/ui/regulations/eregs-vite/src/components/policy-repository/PolicyResults.vue](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-34c8be540f5cf05bd585b4690057fa932410862b0e9679b22c75e724b83838a2)
- [solution/ui/regulations/eregs-vite/src/views/PolicyRepository.vue](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-e3e9af5609a2ada8e770614ffe1d1316b2f9de737fd0cc5e57afa4e912f038a3)

**Cypress Tests:**
[solution/ui/e2e/cypress/e2e/policy-repository.spec.cy.js](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-bdd4c84f360866baec2b33b6d9e5bc54d7b6010c9c168c9065d44d9c4d280427)
tests the ui to see the edit button if user is an admin or manager.
tests the ui to see that non admin will not see the edit button.
tests that the user is navigated to the admin panel for the file that the edit button is clicked on.

**Functional Tests:**
- [solution/backend/regulations/tests/test_admin.py](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1049/files#diff-52a86961c008debd983e6b67219d29bc39675f1f8f1ebb112abd37636fb4f1b9)
tests the changes made in the admin interface.

**Deployment**
Update the github deployment files to create an admin username and password just like it does for a username and password. This allows us to test what an admin has access to vs a non admin. 
- [.github/workflows/deploy-experimental.yml](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1054/files#diff-7a72a63256dac71d3f9b0488baef94eb996b7baab2d06bf6ce213f5079873727)
- [.github/workflows/deploy.yml](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1054/files#diff-28802fbf11c83a2eee09623fb192785e7ca92a3f40602a517c011b947a1822d3)
- [solution/backend/createsu.py](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1054/files#diff-40a651166b9807561475f3302d7c16bc3f46ee43a7739f18f0d544ebabd333aa) 
( create an admin user and add them to e-Regs-Admin group. Create a non admin user and add them to e-Regs-Reader group. So that the cypress tests can check that only none admins can't see the edit link )
[.github/workflows/deploy-experimental.yml](https://github.com/Enterprise-CMCS/cmcs-eregulations/pull/1054/files#diff-7a72a63256dac71d3f9b0488baef94eb996b7baab2d06bf6ce213f5079873727)
(update deploy script to create su after data is seeded so its not written over. Create CYPRESS environment variable for a user that is in the EREGS-READER group.)

---------

**Steps to manually verify this change...**
NONE ADMIN:
1. Sign in through the admin panel. 
Screen Shot 2023-11-13 at 2 18 26 PM
2. click on [users->click on the user name](https://zjl2yo6f5a.execute-api.us-east-1.amazonaws.com/dev1049/admin/auth/user/1/change/)
3. scroll down to Groups and add yourself to EREGS-READER. 
Screen Shot 2023-11-17 at 5 15 37 PM
4. visit[ policy-repository?q=dental](https://zjl2yo6f5a.execute-api.us-east-1.amazonaws.com/dev1049/policy-repository?q=dental). notice that the listed files to the right do not have an edit option.
Screen Shot 2023-11-13 at 2 55 32 PM

ADMIN:

1. Sign in through the admin panel. 
Screen Shot 2023-11-13 at 2 18 26 PM
2. click on [users->click on the user name](https://zjl2yo6f5a.execute-api.us-east-1.amazonaws.com/dev1049/admin/auth/user/1/change/)
3. scroll down to Groups and add yourself to EREGS-ADMIN.  
Screen Shot 2023-11-17 at 5 15 37 PM
4. visit [policy-repository?q=dental](https://zjl2yo6f5a.execute-api.us-east-1.amazonaws.com/dev1049/policy-repository?q=dental). notice that the listed files to the right have an edit option. 
Screen Shot 2023-11-13 at 2 23 56 PM

### CODE REVIEW

**Review by thwalker6 — APPROVED** [review]
lgtm

File: solution/ui/regulations/eregs-vite/src/views/PolicyRepository.vue [hunk 1]
```diff
@@ -60,6 +60,10 @@ const props = defineProps({
         type: Boolean,
         default: false,
     },
+    hasEditableJobCode: {
+      type: Boolean,
+      default: false,
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/views/PolicyRepository.vue:65]: our style is four space indents for JS/Vue files.  This only has two

File: solution/ui/regulations/eregs-vite/src/components/policy-repository/PolicyResults.vue [hunk 2]
```diff
@@ -22,6 +22,10 @@ defineProps({
         type: Object,
         default: () => {},
     },
+    hasEditableJobCode: {
+      type: Boolean,
+      default: false,
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/components/policy-repository/PolicyResults.vue:27]: our style is four space indents for JS/Vue files. This only has two

File: solution/ui/regulations/css/scss/partials/_policy_repository.scss [hunk 3]
```diff
@@ -327,7 +327,11 @@
         .doc-list__document {
             padding: 0.75rem 1rem;
             border-bottom: 1px solid $border_color;
-
+            position: relative;
+            a.edit-button {
+              position: absolute; top: 0; right: 0;
+              text-decoration: none;
```
Comment (PhilR8) [inline solution/ui/regulations/css/scss/partials/_policy_repository.scss:333]: our style for CSS/SCC files is four space indents.  This has two

**Pull request merged**